### PR TITLE
Remove IDisposable contract from IMvcRazorHost

### LINKED ISSUE #4709: Design problem: MvcRazorHost.Dispose
Hi,

MvcRazorHost class injects IChunkTreeCache and assigns it to _chunkTreeCache field (https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Razor.Host/MvcRazorHost.cs#L70). That's a good DI usage, no problem.
But then in MvcRazorHost.Dispose, it disposes the _chunkTreeCache (https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Razor.Host/MvcRazorHost.cs#L348).

Isn't that a design problem? I think a class should not dispose an object if it's injected (not created by this class).

With Dependency injection, IChunkTreeCache's lifetime may not be same as MvcRazorHost. Beside that, DI framework will already dispose IChunkTreeCache when needed. This design can lead to double Dispose of _chunkTreeCache. Maybe this works since MvcRazorHost is somehow singleton, but it's conceptually wrong.

So, a class should only dispose an object if this object is directly created by that class, not injected.

I come across to this situation while debugging my Castle Windsor DI adapter (https://github.com/volosoft/castle-windsor-ms-adapter/tree/master/src).

What do you think?

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #4709

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @NTaylorMullen \ @hikalkan

File: test/Microsoft.AspNetCore.Mvc.Razor.Host.Test/Directives/ChunkInheritanceUtilityTest.cs [hunk 1]
```diff
@@ -32,83 +32,82 @@ @inherits MyBaseType
...
-                var utility = new ChunkInheritanceUtility(host, cache, defaultChunks);
+            var host = new MvcRazorHost(cache, new TagHelperDescriptorResolver(designTime: false));
 
-                // Act
-                var chunkTreeResults = utility.GetInheritedChunkTreeResults(
-                    PlatformNormalizer.NormalizePath(@"Views\home\Index.cshtml"));
+            var utility = new ChunkInheritanceUtility(host, cache, defaultChunks);
 
-                // Assert
-                Assert.Collection(chunkTreeResults,
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.Razor.Host.Test/Directives/ChunkInheritanceUtilityTest.cs]: All of this is just untabbed. The diff is a bit messed up. The split view helps a bit

**Comment by NTaylorMullen** [comment]
:shipit: 

Might be worth an announcement.

**Comment by Eilon** [comment]
Doubt we need an announcement. I'm sure 99.999% of users don't know about any of this. It's plumbing.

**Comment by pranavkm** [comment]
cc @ToddGrun for tooling

**Pull request closed without merging**